### PR TITLE
User Login Implementation (User Stories 13-16)

### PR DESCRIPTION
**Pull request by javier-aguilar**
- User can Login
- User cannot log in with bad credentials
- Users who are logged in already are redirected
- User can log out

### CODE REVIEW

**Comment by mlynch5187** [comment]
All tests passing as well. We utilized BCrypt to make sure passwords were encrypted.

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -17,8 +17,8 @@
   <%= label_tag :email %>
   <%= text_field_tag :email, "#{@user.email}" %>
 
-  <%= label_tag :password_digest %>
-  <%= password_field_tag :password_digest, "#{@user.password_digest}" %>
+  <%= label_tag :password %>
+  <%= password_field_tag :password, "#{@user.password_digest}" %>
```
Comment (reid-andrew) [inline app/views/users/new.html.erb:22]: Just to confirm - the reference to @user.password_digest can stay as is b/c it's pulling out of the DB field, right? Should that even be there? Are we supposed to display the password back to the user at all?

File: spec/features/sessions/destroy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe "As a registered user, merchant, or admin" do
+  before do
+    @user = User.create(name: "Mike Dao",
+                    street_address: "1765 Larimer St",
```
Comment (aperezsantos) [inline spec/features/sessions/destroy_spec.rb:6]: Let's make spacing uniform throughout the users.

**Review by reid-andrew — APPROVED** [review]
This all looks good - I had one question on displaying the pw back to user in edit form which we can address in a refactor if needed.

**Review by aperezsantos — APPROVED** [review]
Awesome work. We can go back and deal with spacing once we get everything up and running.

**Pull request merged**